panel/widgets/tables.py

```diff
@@ -234,7 +234,7 @@ def _get_column_definitions(self, col_names: list[str], df: pd.DataFrame) -> lis
             if isinstance(data, pd.DataFrame):
                 raise ValueError("DataFrame contains duplicate column names.")
 
-            col_kwargs = {}
+            col_kwargs: dict[str, Any] = {}
```

Comment:
In an ideal world this should be replaced with a `TypedDict`

Reply:
In the ideal world you could use `TableColumn` as-is declared in bokeh. But, for many reasons, we had to declare models as dataclasses and you can't get `TypedDict` from a dataclass. Same problem with `Unpack[]` which was supposed to help with `**kwargs`, but we can't use it.

Reply:
This comment was more meant on Panels site of things.